### PR TITLE
Updated User.ts

### PR DESCRIPTION
**Pull request by BaibhavTiwari**
#13 Here's the updated code for the User model.

### CODE REVIEW

**Comment by MitulSonagara** [comment]
@BaibhavTiwari i will review this changes after some time because there is an issue with verification code sending while signup so i need to first fix it and then i can review changes that you made is working properly or not...

**Comment by BaibhavTiwari** [comment]
Ya, sure no worries. 
Take your time

**Comment by BaibhavTiwari** [comment]
@MitulSonagara any updates here?

File: src/model/User.ts [hunk 1]
```diff
@@ -24,8 +25,8 @@ export interface User extends Document {
     verifyCode: string;
     verifyCodeExpiry: Date;
     isVerified: boolean;
-    isAcceptingMessages: boolean; // Consistent naming
-    messages: Message[];
+    isAcceptingMessages: boolean;
+    messages: Schema.Types.ObjectId[];
```
Comment (MitulSonagara) [inline src/model/User.ts]: you can not change the type of message schema to an ObjectId.

File: src/model/User.ts [hunk 2]
```diff
@@ -39,29 +40,43 @@ const UserSchema: Schema<User> = new Schema({
...
+
+UserSchema.methods.isVerifyCodeValid = function () {
+    return this.verifyCodeExpiry > Date.now();
+};
+
+// Hash password before saving
+UserSchema.pre('save', async function (next) {
+    if (!this.isModified('password')) return next();
+    this.password = await bcrypt.hash(this.password, 10);
+    next();
```
Comment (MitulSonagara) [inline src/model/User.ts:84]: Password is already getting stored via hashing process... not need to hash it again

File: src/model/User.ts [hunk 3]
```diff
@@ -39,29 +40,43 @@ const UserSchema: Schema<User> = new Schema({
...
         type: Boolean,
         default: false
     },
-    isAcceptingMessages: { // Consistent naming
+    isAcceptingMessages: {
         type: Boolean,
         default: true
     },
-    messages: [MessageSchema]
+    messages: [{ type: Schema.Types.ObjectId, ref: 'Message' }]
```
Comment (MitulSonagara) [inline src/model/User.ts]: Again messages array has its own type don't need to change it to ObjectId

File: src/model/User.ts (hunk 2 above)
Comment (BaibhavTiwari) [reply src/model/User.ts:84]: Hashing will only occur when the password is modified.

File: src/model/User.ts (hunk 1 above)
Comment (BaibhavTiwari) [reply src/model/User.ts]: done

Comment (BaibhavTiwari) [reply src/model/User.ts]: done

File: src/model/User.ts (hunk 2 above)
Comment (BaibhavTiwari) [reply src/model/User.ts:84]: Hashing will only occur when the password is modified.

Comment (BaibhavTiwari) [reply src/model/User.ts:84]: Hashing will only occur when the password is modified.

File: src/model/User.ts (hunk 3 above)
Comment (BaibhavTiwari) [reply src/model/User.ts]: done

**Review by MitulSonagara — APPROVED** [review]

**Comment by BaibhavTiwari** [comment]
@MitulSonagara ig, you have given the wrong level tag, (lt was level 2 on the issue)

**Pull request merged**